src/sentry/spans/consumers/process/factory.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from typing import Any, Mapping
+
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.run_task import RunTask
+from arroyo.types import BrokerValue, Commit, Message, Partition
+from sentry_kafka_schemas import get_codec
+from sentry_kafka_schemas.codecs import Codec
+from sentry_kafka_schemas.schema_types.snuba_spans_v1 import SpanEvent
+
+from sentry.spans.buffer.redis import RedisSpansBuffer
+from sentry.tasks.spans import process_segment
+
+logger = logging.getLogger(__name__)
+SPAN_SCHEMA: Codec[SpanEvent] = get_codec("snuba-spans")
+
+PROCESS_SEGMENT_DELAY = 2 * 60  # 2 minutes
+
+
+def _deserialize_span(value: bytes) -> Mapping[str, Any]:
+    return SPAN_SCHEMA.decode(value)
+
+
+def process_message(message: Message[KafkaPayload]):
+    assert isinstance(message.value, BrokerValue)
+    try:
+        span = _deserialize_span(message.payload.value)
+        segment_id = span["segment_id"]
```

Comment:
I think we'll want to pack the `segment_id` in the Kafka headers so we don't have to deserialize the whole span just to get the `segment_id`.